src/yang/ietf-bgp-policy.yang

```diff
@@ -47,59 +47,63 @@ module ietf-bgp-policy {
     type enumeration {
       enum ADD {
         description
-          "add the specified communities to the existing
-          community attribute";
+          "Add the specified communities to the existing
+           community attribute";
       }
       enum REMOVE {
         description
-          "remove the specified communities from the
-          existing community attribute";
+          "Remove the specified communities from the
+           existing community attribute";
       }
       enum REPLACE {
         description
-          "replace the existing community attribute with
-          the specified communities. If an empty set is
-          specified, this removes the community attribute
-          from the route.";
+          "Replace the existing community attribute with
+           the specified communities. If an empty set is
+           specified, this removes the community attribute
+           from the route.";
       }
     }
     description
       "Type definition for options when setting the community
-      attribute in a policy action";
+       attribute in a policy action";
   }
 
   typedef bgp-next-hop-type {
     type union {
       type inet:ip-address-no-zone;
       type enumeration {
         enum SELF {
-          description "special designation for local router's own
-          address, i.e., next-hop-self";
+          description
+	    "Special designation for local router's own address, i.e.,
+	     next-hop-self";
         }
       }
     }
     description
-      "type definition for specifying next-hop in policy actions";
+      "Type definition for specifying next-hop in policy actions";
   }
 
   typedef bgp-set-med-type {
     type union {
       type uint32;
       type string {
+	// XXX JMH - this is unbounded.  Also, should the regex be $ terminated?
```

Comment:
Do you have a better suggestion for the regex?

Reply:
The bounds here are ugly.  The underlying type is uint32.  This means that it's bounded to be at most +- MAX_UINT32.  Pick your favorite regex that covers that number range.  It will *certainly* be wrong since it carries the possibility of underflow, but that's a semantic validation impossible to deal with in yang validation.

Reply:
Having gone and read up on MED, I am not sure why should the typedef be a union with a string attribute in it. I think uint32 should be enough. What do you think?

Reply:
The use here is intended to be either "set to this uint32 value" or "set to a string that is plus or minus another value".  MED math is useful, and some implementations also permit some level of multiplication or division with one value and plus/minus with a second.

For this module, +/- would be sufficient.  The challenge is how to model it.

Reply:
How about we have two parameters, the set-med value, and a med-diff value that is +/- of the set-med value?

Reply:
Set, and "do math" tend to be mutually exclusive.

Reply:
I realized that the union had a third type called enumeration that I need to account for. So ignore my previous suggestions. I have enhanced the regex to:

        pattern "^[+-]|[0-9]|[1-8][0-9]|9[0-9]|[1-8][0-9]{2}|" +
	        "9[0-8][0-9]|99[0-9]|[1-8][0-9]{3}|9[0-8][0-9]{2}|" +
	        "99[0-8][0-9]|999[0-9]|[1-8][0-9]{4}|9[0-8][0-9]{3}|" +
	        "99[0-8][0-9]{2}|999[0-8][0-9]|9999[0-9]|" +
	        "[1-8][0-9]{5}|9[0-8][0-9]{4}|99[0-8][0-9]{3}|"
	        "999[0-8][0-9]{2}|9999[0-8][0-9]|99999[0-9]|" +
	        "[1-8][0-9]{6}|9[0-8][0-9]{5}|99[0-8][0-9]{4}|" +
	        "999[0-8][0-9]{3}|9999[0-8][0-9]{2}|99999[0-8][0-9]|" +
	        "999999[0-9]|[1-8][0-9]{7}|9[0-8][0-9]{6}|" +
	        "99[0-8][0-9]{5}|999[0-8][0-9]{4}|9999[0-8][0-9]{3}|" +
	        "99999[0-8][0-9]{2}|999999[0-8][0-9]|9999999[0-9]|" +
	        "[1-8][0-9]{8}|9[0-8][0-9]{7}|99[0-8][0-9]{6}|" +
	        "999[0-8][0-9]{5}|9999[0-8][0-9]{4}|" +
	        "99999[0-8][0-9]{3}|999999[0-8][0-9]{2}|" +
	        "9999999[0-8][0-9]|99999999[0-9]|[1-3][0-9]{9}|" +
	        "4[01][0-9]{8}|42[0-8][0-9]{7}|429[0-3][0-9]{6}|" +
	        "4294[0-8][0-9]{5}|42949[0-5][0-9]{4}|" +
	        "429496[0-6][0-9]{3}|4294967[01][0-9]{2}|" +
	        "42949672[0-8][0-9]|429496729[0-6]";

Reply:
This is off.  The +/- is part of an alternation rather than anchored at the beginning of each sub-expression.  And we're missing the tail anchor, meaning that any gibberish at the end of a valid value is valid.

I know you used a tool to generate.  However, I think this can be simplified.  I'll try to supply a simpler regex later.